src/Guzzle7/Client.php

```diff
@@ -20,26 +20,28 @@
  *
  * @see https://github.com/php-http/guzzle7-adapter/blob/master/src/Client.php
  */
-class Client implements ClientClientInterface
+class Client implements HttpClientInterface
 {
     /**
      * @var ClientInterface
      */
-    private $guzzle;
+    private $client;
 
-    public function __construct(?ClientInterface $guzzle = null)
+    public function __construct(ClientInterface $client = null)
     {
-        if (!$guzzle) {
-            $guzzle = self::buildClient();
+        if (!$client) {
+            $client = self::buildClient();
         }
 
-        $this->guzzle = $guzzle;
+        $this->client = $client;
     }
 
     /**
      * Factory method to create the Guzzle 7 adapter with custom Guzzle configuration.
      *
      * @param array<string, mixed> $config
+     *
+     * @return self
      */
     public static function createWithConfig(array $config): Client
```

Comment:
You did not remove the return type because this class is not supposed to be called?

Reply:
Yes exactly, since Guzzle 7 requires PHP 7.2.5 at least, it shouldn't be called. Do you want to remove them anyway, to be more safe ?

Reply:
My answer above is not really good. Actually, if someone use this lib version (compatible with php5) with Guzzle7, it will mean that the PHP server version will be at least 7.2.5, so this syntax with return type will be supported by the server.